envoy-control-core/src/main/resources/lua/ingress_rbac_logging.lua

```diff
@@ -65,58 +73,80 @@ function is_allowed_client(client_name, clients_allowed_to_all_endpoints)
+        local upstream_request_time = headers:get('x-envoy-upstream-service-time')
+        local status_code = headers:get(':status')
+        local rbac_action = 'shadow_denied'
+        if upstream_request_time == nil and status_code == '403' then
+            rbac_action = 'denied'
         end
-        log_request(lua_metadata, handle, rbac_action)
+        log_request(handle, lua_metadata, jwt_status, rbac_action)
     end
 end
 
-function log_request(lua_metadata, handle, rbac_action)
-    local client_name = lua_metadata["request.info.client_name"] or ""
-    local trusted_client = lua_metadata["request.info.trusted_client"] or false
-    local path = lua_metadata["request.info.path"] or ""
-    local protocol = handle:connection():ssl() == nil and "http" or "https"
-    local method = lua_metadata["request.info.method"] or ""
-    local xff_header = lua_metadata["request.info.xff_header"] or ""
-    local source_ip = string.match(xff_header, '[^,]+$') or ""
-    local request_id = lua_metadata["request.info.request_id"] or ""
-    local status_code = handle:headers():get(":status") or "0"
-    local allowed_client = lua_metadata["request.info.allowed_client"] or false
-    local authority = lua_metadata["request.info.authority"] or ""
-    local lua_destination_authority = lua_metadata["request.info.lua_destination_authority"] or ""
-    handle:logInfo("\nINCOMING_PERMISSIONS { \"method\": \""..method..
-        "\", \"path\": \""..path..
-        "\", \"clientIp\": \""..source_ip..
-        "\", \"clientName\": \""..escape(client_name)..
-        "\", \"trustedClient\": "..tostring(trusted_client)..
-        ", \"authority\": \""..escape(authority)..
-        "\", \"luaDestinationAuthority\": \""..escape(lua_destination_authority)..
-        "\", \"clientAllowedToAllEndpoints\": "..tostring(allowed_client)..
-        ", \"protocol\": \""..protocol..
-        "\", \"requestId\": \""..escape(request_id)..
-        "\", \"statusCode\": "..status_code..
-        ", \"rbacAction\": \""..rbac_action.."\" }")
+function log_request(handle, lua_metadata, jwt_status, rbac_action)
+    local client_name = lua_metadata['request.info.client_name'] or ''
+    local trusted_client = lua_metadata['request.info.trusted_client'] or false
+    local path = lua_metadata['request.info.path'] or ''
+    local protocol = handle:connection():ssl() == nil and 'http' or 'https'
+    local method = lua_metadata['request.info.method'] or ''
+    local xff_header = lua_metadata['request.info.xff_header'] or ''
+    local source_ip = string.match(xff_header, '[^,]+$') or ''
+    local request_id = lua_metadata['request.info.request_id'] or ''
+    local status_code = handle:headers():get(':status') or '0'
+    local allowed_client = lua_metadata['request.info.allowed_client'] or false
+    local authority = lua_metadata['request.info.authority'] or ''
+    local lua_destination_authority = lua_metadata['request.info.lua_destination_authority'] or ''
```

Comment:
how about 'missing' or 'null' instead of empty strings?

Reply:
Created issue https://github.com/allegro/envoy-control/issues/368